codegen: Update ureq to support v3.x

#### Describe the bug/problem

The current ureq code generator produces code using response.into_string(), which is not compatible with ureq 3.x. This method was removed in the major rewrite, leading to compilation errors.

[ureq 3.0.0-rc1 release notes](https://github.com/algesten/ureq/blob/main/CHANGELOG.md#300-rc1)
[ureq 2.12.1 docs](https://docs.rs/ureq/2.12.1/ureq/index.html)
[ureq 3.0.0 docs](https://docs.rs/ureq/3.0.0/ureq/index.html)

In the 2.12.1 docs the method `into_string()` is used but in the next version(3.0.0) the method `into_string()`  is not used.

#### Steps to Reproduce the bug/problem
1. Create a new rust project
2. add ureq v3.x.x to dependencies
3. Run the below code generated from API Dash

```rust
fn main() -> Result<(), ureq::Error> {
    let url = "https://jsonplaceholder.typicode.com/posts";
    let response = ureq::get(url).call()?;

    println!("Response Status: {}", response.status());
    println!("Response: {:?}", response.into_string()?);

    Ok(())
}
```


<img width="935" alt="Image" src="https://github.com/user-attachments/assets/18040937-f4a9-441b-aa7e-589dde4757f7" />


```
error[E0599]: no method named `into_string` found for struct `Response` in the current scope
 --> src/main.rs:6:41
  |
6 |     println!("Response: {:?}", response.into_string()?);
  |                                         ^^^^^^^^^^^ method not found in `Response<Body>`
```

#### Expected behavior
The code should compile and run without any issues.

<img width="509" alt="Image" src="https://github.com/user-attachments/assets/53f43aae-c4ab-48f4-b713-d32286d283a4" />

Image: uses `ureq-2.12.1`

#### Device Info (The device where you encountered this issue)
 - OS: MacOS
 - Version: Sequoia Version 15.3.2 (24D81)

#### Flutter Doctor

```
[✓] Flutter (Channel stable, 3.29.2, on macOS 15.3.2 24D81 darwin-arm64, locale en-US) [2.3s]
    • Flutter version 3.29.2 on channel stable at /Users/udhayxd/tools/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c236373904 (10 days ago), 2025-03-13 16:17:06 -0400
    • Engine revision 18b71d647a
    • Dart version 3.7.2
    • DevTools version 2.42.3

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0) [6.0s]
    • Android SDK at /Users/udhayxd/Library/Android/sdk
    • Platform android-35, build-tools 35.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 21.0.5+-12932927-b750.29)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.2) [2.6s]
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16C5032a
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web [12ms]
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.2) [12ms]
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.5+-12932927-b750.29)

[✓] VS Code (version 1.98.2) [10ms]
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.106.0

[✓] Network resources [1,192ms]
    • All expected network resources are available.

• No issues found!
```

--- a/lib/codegen/rust/ureq.dart
+++ b/lib/codegen/rust/ureq.dart
@@ -5,6 +5,9 @@ import 'package:jinja/jinja.dart' as jj;
 import '../../utils/utils.dart';
 
 class RustUreqCodeGen {
+  final String kJsonImport = """
+use serde_json::json;
+""";
   final String kTemplateStart = """
 {%- if isFormDataRequest -%}
 use std::io::Read;
@@ -23,8 +26,7 @@ fn main() -> Result<(), ureq::Error> {
 
 """;
 
-  String kTemplateJson = """
-    use serde_json::json;
+  String kTemplateJson = """\n
     let payload = json!({{body}});
 
 """;
@@ -90,7 +92,7 @@ multipart/form-data; boundary={{boundary}}''';
     let payload = build_data_list(form_data_items);
 """;
 
-  String kStringRequestBody = """\n        .send_string(payload)?;""";
+  String kStringRequestBody = """\n        .send(payload)?;""";
 
   String kStringRequestForm = """\n        .send_bytes(&payload)?;""";
 
@@ -117,12 +119,16 @@ multipart/form-data; boundary={{boundary}}''';
       String uuid = getNewUuid();
 
       String url = requestModel.url;
-
       var rec = getValidRequestUri(
         url,
         requestModel.enabledParams,
       );
       Uri? uri = rec.$1;
+      var method = requestModel.method;
+      var requestBody = requestModel.body;
+      if (requestModel.bodyContentType == ContentType.json && requestBody?.isNotEmpty == true){
+        result += kJsonImport;
+      }
       if (uri != null) {
         var templateStartUrl = jj.Template(kTemplateStart);
         result += templateStartUrl.render({
@@ -130,9 +136,8 @@ multipart/form-data; boundary={{boundary}}''';
           'isFormDataRequest': requestModel.hasFormData,
           "method": requestModel.method.name.toLowerCase()
         });
-
-        var method = requestModel.method;
-        var requestBody = requestModel.body;
+          
+        
         if (kMethodsWithBody.contains(method) && requestBody != null) {
           var contentLength = utf8.encode(requestBody).length;
           if (contentLength > 0) {

--- a/lib/codegen/rust/ureq.dart
+++ b/lib/codegen/rust/ureq.dart
@@ -30,7 +30,7 @@ fn main() -> Result<(), ureq::Error> {
 """;
 
   String kTemplateHeaders =
-      """\n        {% for key, val in headers -%}.set("{{key}}", "{{val}}"){% if not loop.last %}{{ '\n        ' }}{% endif %}{%- endfor -%}""";
+      """\n        {% for key, val in headers -%}.header("{{key}}", "{{val}}"){% if not loop.last %}{{ '\n        ' }}{% endif %}{%- endfor -%}""";
 
   String kTemplateFormHeaderContentType = '''
 multipart/form-data; boundary={{boundary}}''';

--- a/lib/codegen/rust/ureq.dart
+++ b/lib/codegen/rust/ureq.dart
@@ -24,8 +24,8 @@ fn main() -> Result<(), ureq::Error> {
 """;
 
   String kTemplateJson = """
-
-    let payload = ureq::json!({{body}});
+    use serde_json::json;
+    let payload = json!({{body}});
 
 """;
 

--- a/lib/codegen/rust/ureq.dart
+++ b/lib/codegen/rust/ureq.dart
@@ -100,7 +100,7 @@ multipart/form-data; boundary={{boundary}}''';
 
   final String kStringRequestEnd = """\n
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }

--- a/test/codegen/rust_ureq_codegen_test.dart
+++ b/test/codegen/rust_ureq_codegen_test.dart
@@ -15,7 +15,7 @@ void main() {
         .call()?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -37,7 +37,7 @@ void main() {
         .call()?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -59,7 +59,7 @@ void main() {
         .call()?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -85,7 +85,7 @@ void main() {
         .call()?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -107,7 +107,7 @@ void main() {
         .call()?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -130,7 +130,7 @@ void main() {
         .call()?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -151,7 +151,7 @@ void main() {
         .call()?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -174,7 +174,7 @@ void main() {
         .call()?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -197,7 +197,7 @@ void main() {
         .call()?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -219,7 +219,7 @@ void main() {
         .call()?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -243,7 +243,7 @@ void main() {
         .call()?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -264,7 +264,7 @@ void main() {
         .call()?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -287,7 +287,7 @@ void main() {
         .call()?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -308,7 +308,7 @@ void main() {
         .call()?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -336,7 +336,7 @@ void main() {
         .send_string(payload)?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -366,7 +366,7 @@ void main() {
         .send_json(payload)?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -392,7 +392,7 @@ void main() {
         .send_json(payload)?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -469,7 +469,7 @@ fn main() -> Result<(), ureq::Error> {
         .send_bytes(&payload)?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -547,7 +547,7 @@ fn main() -> Result<(), ureq::Error> {
         .send_bytes(&payload)?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -616,7 +616,7 @@ fn main() -> Result<(), ureq::Error> {
         .send_bytes(&payload)?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -685,7 +685,7 @@ fn main() -> Result<(), ureq::Error> {
         .send_bytes(&payload)?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -761,7 +761,7 @@ fn main() -> Result<(), ureq::Error> {
         .send_bytes(&payload)?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -834,7 +834,7 @@ fn main() -> Result<(), ureq::Error> {
         .send_bytes(&payload)?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -860,7 +860,7 @@ fn main() -> Result<(), ureq::Error> {
         .send_json(payload)?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -888,7 +888,7 @@ fn main() -> Result<(), ureq::Error> {
         .send_json(payload)?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -911,7 +911,7 @@ fn main() -> Result<(), ureq::Error> {
         .call()?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }
@@ -937,7 +937,7 @@ fn main() -> Result<(), ureq::Error> {
         .send_json(payload)?;
 
     println!("Response Status: {}", response.status());
-    println!("Response: {}", response.into_string()?);
+    println!("Response: {}", response.into_body().read_to_string()?);
 
     Ok(())
 }

--- a/test/codegen/rust_ureq_codegen_test.dart
+++ b/test/codegen/rust_ureq_codegen_test.dart
@@ -9,7 +9,8 @@ void main() {
 
   group('GET Request', () {
     test('GET 1', () {
-      const expectedCode = r"""fn main() -> Result<(), ureq::Error> {
+      const expectedCode = r"""
+fn main() -> Result<(), ureq::Error> {
     let url = "https://api.apidash.dev";
     let response = ureq::get(url)
         .call()?;
@@ -103,7 +104,7 @@ void main() {
       const expectedCode = r"""fn main() -> Result<(), ureq::Error> {
     let url = "https://api.github.com/repos/foss42/apidash";
     let response = ureq::get(url)
-        .set("User-Agent", "Test Agent")
+        .header("User-Agent", "Test Agent")
         .call()?;
 
     println!("Response Status: {}", response.status());
@@ -126,7 +127,7 @@ void main() {
     let url = "https://api.github.com/repos/foss42/apidash";
     let response = ureq::get(url)
         .query("raw", "true")
-        .set("User-Agent", "Test Agent")
+        .header("User-Agent", "Test Agent")
         .call()?;
 
     println!("Response Status: {}", response.status());
@@ -170,7 +171,7 @@ void main() {
     let url = "https://api.github.com/repos/foss42/apidash";
     let response = ureq::get(url)
         .query("raw", "true")
-        .set("User-Agent", "Test Agent")
+        .header("User-Agent", "Test Agent")
         .call()?;
 
     println!("Response Status: {}", response.status());
@@ -215,7 +216,7 @@ void main() {
       const expectedCode = r"""fn main() -> Result<(), ureq::Error> {
     let url = "https://api.apidash.dev/humanize/social";
     let response = ureq::get(url)
-        .set("User-Agent", "Test Agent")
+        .header("User-Agent", "Test Agent")
         .call()?;
 
     println!("Response Status: {}", response.status());
@@ -239,7 +240,7 @@ void main() {
     let response = ureq::get(url)
         .query("num", "8700000")
         .query("digits", "3")
-        .set("User-Agent", "Test Agent")
+        .header("User-Agent", "Test Agent")
         .call()?;
 
     println!("Response Status: {}", response.status());
@@ -332,8 +333,8 @@ void main() {
 }"#;
 
     let response = ureq::post(url)
-        .set("content-type", "text/plain")
-        .send_string(payload)?;
+        .header("content-type", "text/plain")
+        .send(payload)?;
 
     println!("Response Status: {}", response.status());
     println!("Response: {}", response.into_body().read_to_string()?);
@@ -351,9 +352,12 @@ void main() {
     });
 
     test('POST 2', () {
-      const expectedCode = r'''fn main() -> Result<(), ureq::Error> {
+      const expectedCode = r'''
+use serde_json::json;
+fn main() -> Result<(), ureq::Error> {
     let url = "https://api.apidash.dev/case/lower";
-    let payload = ureq::json!({
+
+    let payload = json!({
 "text": "I LOVE Flutter",
 "flag": null,
 "male": true,
@@ -381,14 +385,17 @@ void main() {
     });
 
     test('POST 3', () {
-      const expectedCode = r'''fn main() -> Result<(), ureq::Error> {
+      const expectedCode = r'''
+use serde_json::json;
+fn main() -> Result<(), ureq::Error> {
     let url = "https://api.apidash.dev/case/lower";
-    let payload = ureq::json!({
+
+    let payload = json!({
 "text": "I LOVE Flutter"
 });
 
     let response = ureq::post(url)
-        .set("User-Agent", "Test Agent")
+        .header("User-Agent", "Test Agent")
         .send_json(payload)?;
 
     println!("Response Status: {}", response.status());
@@ -465,7 +472,7 @@ fn main() -> Result<(), ureq::Error> {
   
     let payload = build_data_list(form_data_items);
     let response = ureq::post(url)
-        .set("content-type", "multipart/form-data; boundary=test")
+        .header("content-type", "multipart/form-data; boundary=test")
         .send_bytes(&payload)?;
 
     println!("Response Status: {}", response.status());
@@ -542,8 +549,8 @@ fn main() -> Result<(), ureq::Error> {
   
     let payload = build_data_list(form_data_items);
     let response = ureq::post(url)
-        .set("User-Agent", "Test Agent")
-        .set("content-type", "multipart/form-data; boundary=test")
+        .header("User-Agent", "Test Agent")
+        .header("content-type", "multipart/form-data; boundary=test")
         .send_bytes(&payload)?;
 
     println!("Response Status: {}", response.status());
@@ -612,7 +619,7 @@ fn main() -> Result<(), ureq::Error> {
   
     let payload = build_data_list(form_data_items);
     let response = ureq::post(url)
-        .set("content-type", "multipart/form-data; boundary=test")
+        .header("content-type", "multipart/form-data; boundary=test")
         .send_bytes(&payload)?;
 
     println!("Response Status: {}", response.status());
@@ -681,7 +688,7 @@ fn main() -> Result<(), ureq::Error> {
   
     let payload = build_data_list(form_data_items);
     let response = ureq::post(url)
-        .set("content-type", "multipart/form-data; boundary=test")
+        .header("content-type", "multipart/form-data; boundary=test")
         .send_bytes(&payload)?;
 
     println!("Response Status: {}", response.status());
@@ -757,7 +764,7 @@ fn main() -> Result<(), ureq::Error> {
     let response = ureq::post(url)
         .query("size", "2")
         .query("len", "3")
-        .set("content-type", "multipart/form-data; boundary=test")
+        .header("content-type", "multipart/form-data; boundary=test")
         .send_bytes(&payload)?;
 
     println!("Response Status: {}", response.status());
@@ -828,9 +835,9 @@ fn main() -> Result<(), ureq::Error> {
     let response = ureq::post(url)
         .query("size", "2")
         .query("len", "3")
-        .set("User-Agent", "Test Agent")
-        .set("Keep-Alive", "true")
-        .set("content-type", "multipart/form-data; boundary=test")
+        .header("User-Agent", "Test Agent")
+        .header("Keep-Alive", "true")
+        .header("content-type", "multipart/form-data; boundary=test")
         .send_bytes(&payload)?;
 
     println!("Response Status: {}", response.status());
@@ -849,9 +856,11 @@ fn main() -> Result<(), ureq::Error> {
 
   group('PUT Request', () {
     test('PUT 1', () {
-      const expectedCode = r'''fn main() -> Result<(), ureq::Error> {
+      const expectedCode = r'''use serde_json::json;
+fn main() -> Result<(), ureq::Error> {
     let url = "https://reqres.in/api/users/2";
-    let payload = ureq::json!({
+
+    let payload = json!({
 "name": "morpheus",
 "job": "zion resident"
 });
@@ -877,9 +886,12 @@ fn main() -> Result<(), ureq::Error> {
 
   group('PATCH Request', () {
     test('PATCH 1', () {
-      const expectedCode = r'''fn main() -> Result<(), ureq::Error> {
+      const expectedCode = r'''
+use serde_json::json;
+fn main() -> Result<(), ureq::Error> {
     let url = "https://reqres.in/api/users/2";
-    let payload = ureq::json!({
+
+    let payload = json!({
 "name": "marfeus",
 "job": "accountant"
 });
@@ -905,7 +917,8 @@ fn main() -> Result<(), ureq::Error> {
 
   group('DELETE Request', () {
     test('DELETE 1', () {
-      const expectedCode = r"""fn main() -> Result<(), ureq::Error> {
+      const expectedCode = r"""
+fn main() -> Result<(), ureq::Error> {
     let url = "https://reqres.in/api/users/2";
     let response = ureq::delete(url)
         .call()?;
@@ -926,9 +939,12 @@ fn main() -> Result<(), ureq::Error> {
     });
 
     test('DELETE 2', () {
-      const expectedCode = r'''fn main() -> Result<(), ureq::Error> {
+      const expectedCode = r'''
+use serde_json::json;
+fn main() -> Result<(), ureq::Error> {
     let url = "https://reqres.in/api/users/2";
-    let payload = ureq::json!({
+
+    let payload = json!({
 "name": "marfeus",
 "job": "accountant"
 });

